upup/pkg/fi/cloudup/scaleway/cloud.go

```diff
@@ -353,6 +434,45 @@ func (s *scwCloudImplementation) GetClusterVolumes(clusterName string) ([]*insta
 	return volumes.Volumes, nil
 }
 
+func (s *scwCloudImplementation) DeleteLoadBalancer(loadBalancer *lb.LB) error {
+	ipsToRelease := loadBalancer.IP
+
+	// We delete the load-balancer once it's in a stable state
+	_, err := s.lbAPI.WaitForLb(&lb.WaitForLBRequest{
+		LBID:   loadBalancer.ID,
+		Region: s.region,
+	})
+	if err != nil {
+		return fmt.Errorf("error waiting for load-balancer: %w", err)
+	}
+	err = s.lbAPI.DeleteLB(&lb.DeleteLBRequest{
+		Region: s.region,
+		LBID:   loadBalancer.ID,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to delete load-balancer %s: %w", loadBalancer.ID, err)
+	}
+
```

Comment:
I take it deleting an LB also deletes its backend?

Reply:
That's right